closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -348,7 +372,7 @@ int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         len = in.length;
         ibuffer_enc.reset();
 
-        byte[] aad = ((aadBuffer == null || aadBuffer.size() == 0) ? emptyAAD : aadBuffer.toByteArray());
+        byte[] aad = ((aadBuffer.size() == 0 || aadBuffer == null) ? emptyAAD : aadBuffer.toByteArray());
```

Comment:
Calling `aadBuffer.size()` before `aadBuffer == null` may cause NPE.

Reply:
I reverted this change.